instrumentation/spring/spring-boot-autoconfigure-3/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/web/RestClientInstrumentationAutoConfiguration.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.instrumentation.web;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.spring.autoconfigure.internal.SdkEnabled;
+import io.opentelemetry.instrumentation.spring.web.v3_1.SpringWebTelemetry;
+import io.opentelemetry.testing.internal.armeria.client.RestClient;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.web.client.RestClientAutoConfiguration;
+import org.springframework.boot.web.client.RestClientCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Configures {@link RestClient} for tracing.
+ *
+ * <p>Adds Open Telemetry instrumentation to {@link RestClient} beans after initialization
+ */
+@ConditionalOnBean(OpenTelemetry.class)
```

Comment:
```suggestion
@ConditionalOnBean(OpenTelemetry.class)
@ConditionalOnJava(JavaVersion.SEVENTEEN)
```
WIth the good imports...

Reply:
RestClient is part of spring 6, which is jdk 17+: https://github.com/spring-projects/spring-framework/wiki/What's-New-in-Spring-Framework-6.x/d1a3285d5f7658ac716ad4193131102e6f2f50c1#jdk-17-and-jakarta-ee-9-baseline